source/includes/inkling-reference/_schemas.html.md

```diff
@@ -1,265 +1,480 @@
   Int64 | UInt8 | UInt16 | UInt32  | UInt64 | Bool | String
 ```
 
-Select the Syntax tab to show the Inkling set of primitive types which are used in schema declarations. The
-integer suffix indicates the size in bits of the type. Integer types beginning
-with 'U' are unsigned. 
+The Inkling set of primitive types includes numeric, string, and boolean types.
+
+To the right you will see the set of primitive types which are used in schema declarations. 
+
+The integer suffix indicates the size in bits of the type. 
+
+Integer types beginning with 'U' are unsigned. 
+
+`Double` and `Float64` are synonyms.
 
 ###### Structured Types
 
+> Structured Types List
+
 ```inkling--syntax
-structure_type ::= 
-  Luminance | Matrix
+structure_type ::=                      # syntax
+  Luminance 
+  | Matrix
 ```
 
-[//]: # (Are structured types used for simulators ever?)
+```inkling--code
+schema BallLocationSchema               # example
+  Luminance(84, 336) pixels,    
+  Matrix(UInt32, 1, 2) location 
+end
+```
 
+Structured types in Inkling are intended to support common
+Machine Learning types. The Machine Learning types currently supported include
+`Luminance` and `Matrix`.
+  
 Inkling currently supports the types Matrix and Luminance. (This list will be
 expanded.)
 
-See the schema declaration syntax for the complete syntax of structure declaration.
 
-‍
-###### Constrained Types
+###### Constrained Types and Range Expressions
 
 ```inkling--code
-  # Example: Range expression
-
-  schema Schema1
-    Int8 {0:3:10} field   # start:step:stop (all are numeric literals)
+  schema my_schema
+    Int32{1:5} z            # primitive type with type constraint
   end
- 
- # Curly braces delineate the range expression.
- # The values in this range are: (0, 3, 6, 9).
- # Note that the specification of 10 as the stop does not mean
- # 10 must be in the range (because it is a limit, not 
- # necessarily an endpoint).
 ```
 
-Constrained types are supported in schemas and also in [lessons][1]. They are constrained by means of a special type of expression called a range expression.
+A constrained type is a type which is associated with a constraint.
+The type plus constraint is effectively a set definition. 
+A field declared with a
+constrained type can only take values that are a member of the set defined by
+the constraint. In the example to the right, the field `z` can only take on
+values between `1` and `5`. 
+
+Constrained types are supported in schemas and in lesson clauses. Types are
+constrained by means of **range expressions**.
+A range expression on a type has the effect of constraining the values of the type to values defined by the range expression. 
+
+* In a schema the range expression constrains the values in the field. 
+* In lessons the range expression constrains the values of the placeholder being configured.
+
+```inkling--code
+schema MyOutput 
+   UInt8 {0,1,2,3,4}    label,       # a list of UInt8 values
```

Comment:
All real code samples that I see don't have a space between the constraint and the type. We should try and be consistent. Is the space needed or not?

Reply:
yes, true. Spaces are not required, though they are allowed. I like having spaces here because

- it shows they are allowed.

- more whitespace makes it a little easier to read